api/src/controllers/internal/processProgram.ts

```diff
@@ -49,7 +49,7 @@ export const processProgram = async (
     await moveFile(codePath + '.bkp', codePath)
 
     // we now need to poll the session status
-    while (!session.completed) {
+    while (session.state === SessionState.completed) {
```

Comment:
Can you check this condition again? I think it should be `session.state !== SessionState.completed`

Reply:
you are right it should be `!==`